blockchain/ethereum/internal/chain.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2020 - for information on the respective copyright owner
+// see the NOTICE file and/or the repository at
+// https://github.com/direct-state-transfer/dst-go
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+
+	ethchannel "perun.network/go-perun/backend/ethereum/channel"
+	ethwallet "perun.network/go-perun/backend/ethereum/wallet"
+	"perun.network/go-perun/channel"
+	"perun.network/go-perun/wallet"
+)
+
+// ChainBackend provides ethereum specific contract backend functionality.
+type ChainBackend struct {
+	// Cb is the instance of contract backend that will be used for all on-chain communications.
+	Cb *ethchannel.ContractBackend
+	// TxTimeout is the max time to wait for confirmation of transactions on blockchain.
+	// If this expires, a transactions is considered failed.
+	// Use sufficiently large values when connecting to mainnet.
+	TxTimeout time.Duration
+}
+
+// NewFunder initializes and returns an instance of ethereum funder.
+func (cb *ChainBackend) NewFunder(assetAddr wallet.Address) channel.Funder {
```

Comment:
`NewFunder` and `NewAdjudicator` are untested right now.

Reply:
Yes, I am not sure what to test. Because, the only expected behavior of these functions is that they return valid interfaces. 

Due to special structuring (as described in the package docs) of this package, all its method signature should not contain types defined in `go-ethereum` or `go-perun/backend/ethereum`, the return type is already an interface.

If it was otherwise returning a struct type, i would have added test to check if the structs implement corresponding interfaces.

Do you have suggestion for testing these two APIs ?